### PR TITLE
Pv2 1389 enum conversion problem in identified removed learner aim

### PR DESCRIPTION
**Pull request by davehowlett**

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/RefundRemovedLearningAimProcessor.cs [hunk 1]
```diff
@@ -41,31 +41,41 @@ public async Task<ReadOnlyCollection<PeriodisedRequiredPaymentEvent>> RefundLear
             var historicPayments = cacheItem.Value.Select(mapper.Map<PaymentHistoryEntity, Payment>).ToList();
             logger.LogDebug($"Got {historicPayments.Count} historic payments. Now generating refunds per transaction type.");
 
+            // Now restricted the historic payments from the cache to those for the transaction type in that particular group.
             var requiredPaymentEvents = historicPayments.GroupBy(historicPayment => historicPayment.TransactionType)
-                .SelectMany(group => CreateRefundPayments(identifiedRemovedLearningAim, historicPayments, group.Key, cacheItem))
+                .SelectMany(group => CreateRefundPayments(
+                    identifiedRemovedLearningAim,
```
Comment (carlbm) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Processors/RefundRemovedLearningAimProcessor.cs]: Does it make sense to 
`historicPayments
.SelectMany(x => CreateRefundPayments(...)
.GroupBu(x => x.TransactionType)
`
?

File: src/SFA.DAS.Payments.RequiredPayments.Application/Processors/RefundRemovedLearningAimProcessor.cs [hunk 2]
```diff
@@ -41,31 +41,41 @@ public async Task<ReadOnlyCollection<PeriodisedRequiredPaymentEvent>> RefundLear
...
 
                     if (historicPayment == null)
                         throw new InvalidOperationException($"Cannot find historic payment with price episode identifier: {refund.payment.PriceEpisodeIdentifier} for period {refund.deliveryPeriod}.");
                     
                     var requiredPaymentEvent = requiredPaymentEventFactory.Create(refund.payment.EarningType, transactionType);
-
+                    if (requiredPaymentEvent == null)
+                    {
+                        // This shouldn't now happen as the transaction type in the history should match the one in the cache
+                        return null;
```
Comment (carlbm) [inline src/SFA.DAS.Payments.RequiredPayments.Application/Processors/RefundRemovedLearningAimProcessor.cs]: Can we log a warning here? If it shouldn't happen then the warning shouldn't be fired that often

Comment (CarlHA) [reply src/SFA.DAS.Payments.RequiredPayments.Application/Processors/RefundRemovedLearningAimProcessor.cs]: logging added

**Comment by samuel-dev-dba** [comment]
closing obsolete PRs

**Pull request closed without merging**